### PR TITLE
Add support for limiting depth of `walk_dir`

### PR DESCRIPTION
**Pull request by panhania**

### CODE REVIEW

File: crates/rrg/src/fs.rs [hunk 1]
```diff
@@ -81,48 +68,14 @@ pub fn walk_dir<P: AsRef<Path>>(root: P) -> std::io::Result<WalkDir> {
...
-/// ```
-pub fn list_dir<P: AsRef<Path>>(path: P) -> std::io::Result<ListDir> {
-    let iter = std::fs::read_dir(path)?;
-
-    Ok(ListDir {
-        iter: iter,
-    })
-}
-
 /// Iterator over entries in all subdirectories.
```
Comment (s-westphal) [inline crates/rrg/src/fs.rs:79]: add "until `max_depth`"?

File: crates/rrg/src/fs.rs [hunk 2]
```diff
@@ -135,13 +88,39 @@ pub fn list_dir<P: AsRef<Path>>(path: P) -> std::io::Result<ListDir> {
...
+    /// assert!(!paths.contains(&"/usr/lib".into()));
+    /// ```
+    pub fn with_max_depth(mut self, max_depth: u32) -> WalkDir {
+        assert!(max_depth > 0);
+
+        self.max_depth = max_depth;
+        self
+    }
+
     #[cfg(target_family = "unix")]
```
Comment (s-westphal) [inline crates/rrg/src/fs.rs:124]: What happens on macos?

File: crates/rrg/src/fs.rs [hunk 3]
```diff
@@ -203,7 +189,21 @@ impl std::iter::Iterator for ListDir {
     type Item = std::io::Result<Entry>;
```
Comment (s-westphal) [inline crates/rrg/src/fs.rs:189]: Is this needed?

**Review by s-westphal — APPROVED** [review]

File: crates/rrg/src/fs.rs (hunk 1 above)
Comment (panhania) [reply crates/rrg/src/fs.rs:79]: `max_depth` is a private field of the struct and docs are for public consumption, it should not leak implementation details.

However, I added a paragraph with reference to the `with_max_depth` method.

File: crates/rrg/src/fs.rs (hunk 2 above)
Comment (panhania) [reply crates/rrg/src/fs.rs:126]: `unix` option of [`target_family`] covers both macOS and Linux, as opposed to [`target_os`] that disambiguates between the two.

[`target_family`]: https://doc.rust-lang.org/reference/conditional-compilation.html#target_family
[`target_os`]: https://doc.rust-lang.org/reference/conditional-compilation.html#target_os

File: crates/rrg/src/fs.rs (hunk 3 above)
Comment (panhania) [reply crates/rrg/src/fs.rs:191]: I am not sure what you mean by "needed". `Item` [is an associated type][1] on the `Iterator` trait.

[1]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#required-associated-types

Comment (s-westphal) [reply crates/rrg/src/fs.rs:191]: Oh I see, thank you, I thought it is a [type alias](https://doc.rust-lang.org/reference/items/type-aliases.html) which is then not used.

**Review by s-westphal — APPROVED** [review]

File: crates/rrg/src/fs.rs (hunk 3 above)
Comment (panhania) [reply crates/rrg/src/fs.rs:191]: Ah, okay. Just so you know then: if you see `type ... = ...;` declaration in trait implementation block (`impl ... for ... { ... }`), it _has to_ be an associated type, traditional aliases are not allowed there.

**Pull request merged**